File: app/src/main/java/io/github/ovso/righttoknow/data/network/VideoRequest.java
package io.github.ovso.righttoknow.data.network;

import android.text.TextUtils;
import io.github.ovso.righttoknow.Security;
import io.github.ovso.righttoknow.data.KeyName;
import io.github.ovso.righttoknow.data.network.model.video.Search;
import io.reactivex.Single;
import java.util.HashMap;
import java.util.Map;
import javax.inject.Inject;
import okhttp3.Headers;

public class VideoRequest extends BaseRequest<VideoService> {

  @Inject public VideoRequest() {

  }

  @Override protected Class<VideoService> getApiClass() {
    return VideoService.class;
  }

  @Override protected Headers createHeaders() {

    return new Headers.Builder().build();
  }

  @Override protected String getBaseUrl() {
    return ApiEndPoint.SEARCH.getUrl();
  }

  public Single<Search> getResult(String q, String pageToken) {
    return getApi().getResult(createQueryMap(q, pageToken));
  }

  private Map<String, Object> createQueryMap(String q, String pageToken) {
    Map<String, Object> queryMap = new HashMap<>();
    if (!TextUtils.isEmpty(q)) {
      queryMap.put(KeyName.Q.get(), q);
    }
    queryMap.put(KeyName.MAX_RESULTS.get(), 50);
    queryMap.put(KeyName.ORDER.get(), "date");
    queryMap.put(KeyName.TYPE.get(), "video");
    queryMap.put(KeyName.VIDEO_SYNDICATED.get(), "true");
    queryMap.put(KeyName.KEY.get(), Security.GOOGLE_API_KEY.getValue());
    queryMap.put(KeyName.PART.get(), "snippet");
    if (!TextUtils.isEmpty(pageToken)) {
      queryMap.put(KeyName.PAGE_TOKEN.get(), pageToken);
    }
    return queryMap;
  }
}

File: app/src/main/java/io/github/ovso/righttoknow/data/network/VideoRequest.kt
package io.github.ovso.righttoknow.data.network

import android.text.TextUtils
import io.github.ovso.righttoknow.Security
import io.github.ovso.righttoknow.data.KeyName
import io.github.ovso.righttoknow.data.network.model.video.Search
import io.reactivex.Single
import okhttp3.Headers
import java.util.*
import kotlin.collections.HashMap

class VideoRequest : BaseRequest<VideoService>() {
  override val apiClass: Class<VideoService>
    protected get() = VideoService::class.java

  override fun createHeaders(): Headers? {
    return Headers.Builder().build()
  }

  override val baseUrl: String
    protected get() = ApiEndPoint.SEARCH.url

  fun getResult(q: String, pageToken: String?): Single<Search> {
    return api.getResult(createQueryMap(q, pageToken))
  }

  private fun createQueryMap(q: String, pageToken: String?): HashMap<String, Any> {
    val queryMap: HashMap<String, Any> = HashMap()
    if (!TextUtils.isEmpty(q)) {
      queryMap[KeyName.Q.get()] = q
    }
    queryMap[KeyName.MAX_RESULTS.get()] = 50
    queryMap[KeyName.ORDER.get()] = "date"
    queryMap[KeyName.TYPE.get()] = "video"
    queryMap[KeyName.VIDEO_SYNDICATED.get()] = "true"
    queryMap[KeyName.KEY.get()] = Security.GOOGLE_API_KEY.value
    queryMap[KeyName.PART.get()] = "snippet"
    pageToken?.let {
      queryMap[KeyName.PAGE_TOKEN.get()] = it
    }
    return queryMap
  }
}
